src/mc.rs

```diff
@@ -1229,3 +1216,1069 @@ pub unsafe fn resize_rust<BD: BitDepth>(
+        ptrdiff_t,
+        libc::c_int,
+        libc::c_int,
+        libc::c_int,
+        libc::c_int,
+        libc::c_int,
+        libc::c_int,
+        libc::c_int,
+    ) -> (),
+>;
+pub type warp8x8t_fn = Option<
+    unsafe extern "C" fn(
+        *mut int16_t,
+        ptrdiff_t,
+        *const DynPixel,
+        ptrdiff_t,
+        *const int16_t,
+        libc::c_int,
+        libc::c_int,
+        libc::c_int,
+    ) -> (),
+>;
+pub type avg_fn = Option<
+    unsafe extern "C" fn(
+        *mut DynPixel,
+        ptrdiff_t,
+        *const int16_t,
+        *const int16_t,
+        libc::c_int,
+        libc::c_int,
+        libc::c_int,
+    ) -> (),
+>;
+pub type w_avg_fn = Option<
+    unsafe extern "C" fn(
+        *mut DynPixel,
+        ptrdiff_t,
+        *const int16_t,
+        *const int16_t,
+        libc::c_int,
+        libc::c_int,
+        libc::c_int,
+        libc::c_int,
+    ) -> (),
+>;
+pub type mask_fn = Option<
+    unsafe extern "C" fn(
+        *mut DynPixel,
+        ptrdiff_t,
+        *const int16_t,
+        *const int16_t,
+        libc::c_int,
+        libc::c_int,
+        *const uint8_t,
+        libc::c_int,
+    ) -> (),
+>;
+pub type w_mask_fn = Option<
+    unsafe extern "C" fn(
+        *mut DynPixel,
+        ptrdiff_t,
+        *const int16_t,
+        *const int16_t,
+        libc::c_int,
+        libc::c_int,
+        *mut uint8_t,
+        libc::c_int,
+        libc::c_int,
+    ) -> (),
+>;
+pub type blend_fn = Option<
+    unsafe extern "C" fn(
+        *mut DynPixel,
+        ptrdiff_t,
+        *const DynPixel,
+        libc::c_int,
+        libc::c_int,
+        *const uint8_t,
+    ) -> (),
+>;
+pub type blend_dir_fn = Option<
+    unsafe extern "C" fn(*mut DynPixel, ptrdiff_t, *const DynPixel, libc::c_int, libc::c_int) -> (),
+>;
+pub type emu_edge_fn = Option<
+    unsafe extern "C" fn(
+        intptr_t,
+        intptr_t,
+        intptr_t,
+        intptr_t,
+        intptr_t,
+        intptr_t,
+        *mut DynPixel,
+        ptrdiff_t,
+        *const DynPixel,
+        ptrdiff_t,
+    ) -> (),
+>;
+pub type resize_fn = Option<
+    unsafe extern "C" fn(
+        *mut DynPixel,
+        ptrdiff_t,
+        *const DynPixel,
+        ptrdiff_t,
+        libc::c_int,
+        libc::c_int,
+        libc::c_int,
+        libc::c_int,
+        libc::c_int,
+        libc::c_int,
+    ) -> (),
+>;
+#[repr(C)]
+pub struct Dav1dMCDSPContext {
+    pub mc: [mc_fn; 10],
+    pub mc_scaled: [mc_scaled_fn; 10],
+    pub mct: [mct_fn; 10],
+    pub mct_scaled: [mct_scaled_fn; 10],
+    pub avg: avg_fn,
+    pub w_avg: w_avg_fn,
+    pub mask: mask_fn,
+    pub w_mask: [w_mask_fn; 3],
+    pub blend: blend_fn,
+    pub blend_v: blend_dir_fn,
+    pub blend_h: blend_dir_fn,
+    pub warp8x8: warp8x8_fn,
+    pub warp8x8t: warp8x8t_fn,
+    pub emu_edge: emu_edge_fn,
+    pub resize: resize_fn,
+}
+
+macro_rules! filter_fns {
+    ($mc_kind:ident, $type_h:expr, $type_v:expr) => {
```

Comment:
I didn't realize there was another `$mc_kind`.  This one is different so it's kinda confusing.  What about `$type_names` for this one, since it's a combination of the names of `$type_h` and `$type_v`?

Reply:
I'm gonna leave this one `mc_kind` now that the other one has been changed to `$fn_kind`.